CMakeLists.txt

```diff
@@ -104,6 +97,21 @@ add_custom_command(
           ${yaml_SRC}
           ${templates_SRC})
 
+set(model-GENERATED_SRC ${CMAKE_CURRENT_BINARY_DIR}/src/UHDM.capnp
+                        ${CMAKE_CURRENT_BINARY_DIR}/src/UHDM.capnp.h
+                        ${CMAKE_CURRENT_BINARY_DIR}/src/UHDM.capnp.c++)
+foreach(header_file ${model-GENERATED_SRC})
+  set_source_files_properties(${header_file} PROPERTIES GENERATED TRUE)
+endforeach(header_file ${model-GENERATED_SRC})
+
+# Q: how can I tell capnp_generate_cpp that UHDM.capnp is generatede first?
+# DEPENDS does not work.
+capnp_generate_cpp(CAPNP_SOURCE CAPNP_HEADER
+                  ${CMAKE_CURRENT_BINARY_DIR}/src/UHDM.capnp
+                  DEPENDS ${model-GENERATED_UHDM})
```

Comment:
It is not clear to me how to tell cmake that we really first have to run the command that gernrates the UHDM before this can be run.

Reply:
`capnp_generate_cpp()` is a function and there is no explicit mechanism in cmake to add dependency between targets and function calls. The only alternative is to wrap the function call itself in a custom command and use that command to setup dependencies. Here's an [example](https://stackoverflow.com/questions/26024235/how-to-call-a-cmake-function-from-add-custom-target-command)

Also, note that function execution happens at build configuration time and will be delayed until the end automatically if the outputs of it have no explicit dependencies.

Hope this helps!!

Reply:
Thanks. After fiddling with a `-P` sub-call to cmake, which resulted in its own set of problems (e.g. can't use `FindPackage(CapnProto)` in there), I resorted in making it just a `add_custom_command()`